packages/common/src/utils/updatePlaylistArtwork.ts

```diff
@@ -0,0 +1,93 @@
+import { isEqual } from 'lodash'
+
+import { Collection } from 'models/Collection'
+import { SquareSizes } from 'models/ImageSizes'
+import { Track } from 'models/Track'
+import { AudiusBackend } from 'services/audius-backend'
+
+import { Nullable } from './typeUtils'
+
+type ArtworkActions = {
+  added?: Track
+  removed?: Track
+  reordered?: Track[]
+  regenerate?: boolean
+}
+
+type Context = {
+  audiusBackend: AudiusBackend
+  generateImage: (urls: string[]) => Promise<{ url: string; file: File }>
+}
+
+export const updatePlaylistArtwork = async (
+  collection: Collection,
+  tracks: Track[],
+  actions: ArtworkActions,
+  context: Context
+) => {
+  const { is_image_autogenerated } = collection
+  if (!is_image_autogenerated && !actions.regenerate) {
+    return collection
+  }
+
+  let tracksForImage: Nullable<Track[]> = null
+
+  if (actions.added) {
+    if (tracks.length === 0) {
+      tracksForImage = [actions.added]
+    } else if (tracks.length === 3) {
+      tracksForImage = [...tracks, actions.added]
+    }
+  } else if (actions.removed) {
+    const { removed } = actions
+    const removedIndex = tracks.findIndex(
+      (track) => track.track_id === removed.track_id
+    )
+    if (removedIndex === -1) {
+      // continue
```

Comment:
what case is this?

Reply:
oh is this just the default value passed when nothing is removed?

Reply:
yeah in case we dont get a match, just gonna carry on :)